task.aspx.cs

```diff
@@ -9,9 +12,275 @@ namespace LogBook
+
+                if (dt.Rows.Count >= 1)
+                {
+                    txtTaskId.Text = dt.Rows[0][1].ToString();
+                }
+                else
+                {
+                    Response.Write("<script>alert('Invalid Task ID');</script>");
+                }
+            }
+            catch (Exception ex)
+            {
+                Response.Write("<script>alert(''" + ex.Message + "');</script>");
+                
+            }
+        }
+
+        /// <summary>
+        /// Delete a task using an ID.
+        /// If succesful, the table will be cleared.
+        /// </summary>
+        void DeleteTask()
+        {
+            try
+            {
+                SqlConnection con = new SqlConnection(strCon);
+                if (con.State == ConnectionState.Closed)
+                {
+                    con.Open();
+                }
+
+                SqlCommand cmd = new SqlCommand("DELETE from task_detail_tbl WHERE task_id='" + txtTaskId.Text.Trim() + "'", con);
+
+                cmd.ExecuteNonQuery();
+                con.Close();
+                Response.Write("<script>alert('Task Deleted Successfully');</script>");
+                ClearForm();
+                GridView1.DataBind();
+
+            }
+            catch (Exception ex)
+            {
+                Response.Write("<script>alert(''" + ex.Message + "');</script>");
+            }
+        }
+
+        /// <summary>
+        /// Update a task using an ID.
+        /// If succesful, the table will be updated.
+        /// </summary>
+        void UpdateTask()
+        {
+            try
+            {
+                SqlConnection con = new SqlConnection(strCon);
+                if (con.State == ConnectionState.Closed)
+                {
+                    con.Open();
+                }
+
+                SqlCommand cmd = new SqlCommand("UPDATE task_detail_tbl SET task_name=@task_name WHERE task_id='" + txtTaskId.Text.Trim()+"'", con);
+                cmd.Parameters.AddWithValue("@task_name", txtTaskName.Text.Trim());
+
+                cmd.ExecuteNonQuery();
+                con.Close();
+                Response.Write("<script>alert('Task updated Successfully');</script>");
+                ClearForm();
+                GridView1.DataBind();
+
+            }
+            catch (Exception ex)
+            {
+                Response.Write("<script>alert(''" + ex.Message + "');</script>");
+            }
+        }
+
+        /// <summary>
+        /// Add a nre task using an ID.
+        /// If succesful, the table will be added.
+        /// </summary>
+        void AddNewTask()
+        {
+            string taskid = txtTaskId.Text.Trim();
+            string taskname = txtTaskName.Text.Trim();
+            string task = txtTask.Text.Trim(); 
+            string challenges = txtChallenges.Text.Trim(); 
+            string overcome = txtOvercome.Text.Trim(); 
+            string learning_experience = txtLearningX.Text.Trim(); 
```

Comment:
### Naming Conventions. 

Be consistent with your choice of naming conventions. 
If you are using _, then use _ for variables with more than one word. 
```suggestion
            string task_id= txtTaskId.Text.Trim();
            string task_name= txtTaskName.Text.Trim();
            string task = txtTask.Text.Trim(); 
            string challenges = txtChallenges.Text.Trim(); 
            string overcome = txtOvercome.Text.Trim(); 
            string learning_exp = txtLearningX.Text.Trim(); 
```
or following the proper C# naming convention,

```suggestion
            string taskID= txtTaskId.Text.Trim();
            string taskName= txtTaskName.Text.Trim();
            string task = txtTask.Text.Trim(); 
            string challenges = txtChallenges.Text.Trim(); 
            string overcome = txtOvercome.Text.Trim(); 
            string learningExp = txtLearningX.Text.Trim(); 
```